pkg/apis/v1alpha1/subnetport_types.go

```diff
@@ -0,0 +1,83 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetPortSpec defines the desired state of SubnetPort.
+type SubnetPortSpec struct {
+	// Subnet defines the parent Subnet name of the SubnetPort.
+	Subnet string `json:"subnet,omitempty"`
+	// SubnetSet defines the parent SubnetSet name of the SubnetPort.
+	SubnetSet string `json:"subnetset,omitempty"`
+	// Attachment refers to the virtual machine which the SubnetPort is attached.
+	Attachment corev1.TypedLocalObjectReference `json:"attachment,omitempty"`
+}
+
+// SubnetPortStatus defines the observed state of SubnetPort.
+type SubnetPortStatus struct {
+	// Conditions describes current state of SubnetPort.
+	Conditions []SubnetPortCondition `json:"conditions,omitempty"`
+	// InterfaceID describes the attachment VIF ID owned by the SubnetPort in NSX-T.
+	InterfaceID string `json:"interfaceID,omitempty"`
+	// IPAddresses describes the IP addresses of the SubnetPort.
+	IPAddresses []SubnetPortIPAddress `json:"ipAddresses,omitempty"`
+	// MacAddress describes the MAC address of the SubnetPort.
+	MacAddress string `json:"macAddress,omitempty"`
+	// ProviderStatus defines the NSX-T resource provider status.
+	ProviderStatus SubnetPortProviderStatus `json:"providerStatus,omitempty"`
```

Comment:
I feel we can just put LogicalSwitchID here, and need not a "SubnetPortProviderStatus" struct. We do not have multiple providers right.

Reply:
make sense, done